crates/iota-core/src/transaction_input_loader.rs

```diff
@@ -161,17 +167,20 @@ impl TransactionInputLoader {
                     fetches.push((i, input));
                 }
                 InputObjectKind::SharedMoveObject { id, .. } => {
-                    let shared_locks = shared_locks_cell.get_or_try_init(|| {
-                        Ok::<HashMap<ObjectID, SequenceNumber>, IotaError>(
+                    let shared_locks = shared_locks_cell
+                        .get_or_init(|| {
                             shared_lock_store
-                                .get_shared_locks(tx_key)?
-                                .into_iter()
-                                .collect(),
-                        )
-                    })?;
-                    // If we can't find the locked version, it means
-                    // 1. either we have a bug that skips shared object version assignment
-                    // 2. or we have some DB corruption
+                                .get_shared_locks(tx_key)
+                                .expect("loading shared locks should not fail")
+                                .map(|locks| locks.into_iter().collect())
+                        })
+                        .as_ref()
+                        .unwrap_or_else(|| {
+                            // _tx_lock is held, so this should not happen
+                            panic!("Failed to get shared locks for transaction {tx_key:?}");
```

Comment:
Should we add a TODO that "fatal" should be used here in the future?

Reply:
Thanks! Fixed at [bec3e9b](https://github.com/iotaledger/iota/pull/4757/commits/bec3e9b139251a8fb5a5d6595c681c517ae9030d).

Reply:
@bingyanglin `fatal!` shoud be available now, so please update